standard-covoiturage_openapi.yaml

```diff
@@ -307,79 +392,163 @@ components:
+        - type: object
+          required:
+            - passenger
+          properties:
+            passenger:
+              $ref: '#/components/schemas/User'
+
+    Journey:
+      title: Journey
+      allOf:
+        - $ref: '#/components/schemas/Trip'
+        - type: object
+          required:
+            - passengerPickupDate
+            - type
+            - webUrl
+          properties:
+            id:
+              type: string
+              minLength: 1
+              maxLength: 255
+              description: Journey's id. It MUST be unique for a given operator.
+            passengerPickupDate:
+              type: number
+              description: |-
+                Passenger pickup datetime as a UNIX UTC timestamp in seconds.
+              format: long
+            driverDepartureDate:
+              type: number
+              description: |-
+                Driver departure datetime as a UNIX UTC timestamp in seconds.
+              format: long
+            webUrl:
+              type: string
+              description: URL of the journey on the webservice provider platform.
+            type:
+              type: string
+              description: |-
+                Type of journey. A dynamic journey is happening in real time.
+              enum:
+              - PLANNED
+              - DYNAMIC
+              - LINE
+            deepLink:
+              $ref: '#/components/schemas/DeepLink'
+
+    DriverJourney:
+      allOf:
+        - $ref: '#/components/schemas/DriverTrip'
+        - $ref: '#/components/schemas/Journey'
+        - type: object
+          properties:
+            availableSeats:
+              type: integer
+              description: Available seats in the car.
+            price:
+              $ref: '#/components/schemas/Price'
 
     PassengerJourney:
       allOf:
+        - $ref: '#/components/schemas/PassengerTrip'
         - $ref: '#/components/schemas/Journey'
         - type: object
           required:
-            - passenger
             - driverDepartureDate
           properties:
-            passenger:
-              $ref: '#/components/schemas/User'
             requestedSeats:
               type: integer
-              description: If a passenger journey, requested seats.
+              description: Requested seats by the passenger.
+
+    Schedule:
+      type: object
+      properties:
+        passengerPickupDay:
+          type: string
+          description: Day of week of the passenger pick-up.
+          enum:
+            - mon
+            - tue
+            - wed
+            - thu
+            - fri
+            - sat
+            - sun
+        passengerPickupTimeOfDay:
+          description: |-
+            Passenger pick-up time of day represented as
+            [RFC3339 partial-time](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6).
+          type: string
+          format: partial-time
+        journeyIds:
```

Comment:
As far as I understand, we get a list of journeyIDs without corresponding dates. How did you imagine the flow on this ? Wouldn't it be easier to retrieve an array of "journeyID + date" instead of just journeyID to avoid useless processing on the MaaS side just to retrieve dates correspondence with IDs ? Or was there a specific reason you just wanted IDs without further information ?

Reply:
@adelcasse Two reasons I designed it like this : 
- we don't have widespread UX in place to take example on, so I had to imagine. So I imagined that in most cases, people would be interested at first on which driver/passenger could match their habits (and "how" => days, time of day, itinerary, walking time, etc.) and subsequently we would focus on specific instance of journeys.
- adding anything else than `journeyIds` in `schedules` [a] would require to introduce another concept/model (something like  "JourneySchedules'), making the specification and the concepts harder to understand [b] unclear boundaries between this new concept and a `Journey`, with the risk that with time, this new concept would include more and more of a `journey`.

In all cases, `/journeys` endpoint (with the possibility to retrieve many `journeys`, even from multiple drivers / passengers) is here to support "index" views where more info from journeys would be needed while displaying `trips`.

That beeing said, I am open to suggestions :)

Reply:
Hi @ccyrille, thanks for the proposal. I'm note sure to undestand exactly the meaning of Schedule:
Do you expect at most one `Schedule` for each weekday?
Is `journeyIds` an array so that it can contain journeys for several weeks in advance?

I think I agree with @adelcasse and actually we could go further, as you mention in your last comment: create an object `JourneySchedule` which contains `id`, `passengerPickupDate`, `driverDepartureDate`, `webUrl`, `type`, `deeplink`
`Journey` could reference `JourneySchedule`, and here we could return an array of `JourneySchedule`
Would that make sense?